mslib/mscolab/file_manager.py

```diff
@@ -250,6 +254,42 @@ def modify_user(self, user, attribute=None, value=None, action=None):
             db.session.commit()
         return True
 
+    def save_user_profile_image(self, user_id, image_file):
+        """
+        Save the user's profile image path to the database.
+        """
+        upload_folder = mscolab_settings.UPLOAD_FOLDER
+        file_token = secrets.token_urlsafe(20)
+
+        with fs.open_fs('/') as home_fs:
+            file_dir = fs.path.join(upload_folder, 'profile')
+            # Checking if directory exists. Create the directory if it does not exist
+            if '\\' not in file_dir:
+                if not home_fs.exists(file_dir):
+                    home_fs.makedirs(file_dir)
+            else:
+                file_dir = file_dir.replace('\\', '/')
+                if not os.path.exists(file_dir):
+                    os.makedirs(file_dir)
+
+            # Creating unique and secure filename
+            file_name, file_ext = image_file.filename.rsplit('.', 1)
+            file_name = f'{user_id}-{time.strftime("%Y%m%dT%H%M%S")}-{file_token}.{file_ext}'
+            file_name = secure_filename(file_name)
+
+            # Saving the file to the filepath
+            file_path = fs.path.join(file_dir, file_name)
+            image_file.save(file_path)
+
+            # Storing the file path to database
+            user = User.query.get(user_id)
+            if user:
+                user.profile_image_path = file_path
```

Comment:
I don't think we should store the absolute path in the database. Instead, we should only put the path relative to the configured profile image folder in there, to support easy relocation in case of configuration changes. Also, it's less chars to store.